File: akka-benchmark/src/main/java/io/sarl/akka/eventstream/PublisherActor.java
package io.sarl.akka.eventstream;

import akka.actor.AbstractActor;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class PublisherActor extends AbstractActor {
    static public Props props() {
        return Props.create(PublisherActor.class, () -> new PublisherActor());
    }


    private LoggingAdapter log = Logging.getLogger(getContext().getSystem(), this);

    public static class TriggerEmit {
        //
    }

    public PublisherActor() {
        //
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(TriggerEmit.class, event -> {
                    getContext().getSystem().eventStream().publish(new Message("This is a message"));
                })
                .build();
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/eventstream/PublisherActor.kt
package io.sarl.akka.eventstream

import akka.actor.AbstractActor
import akka.actor.Props
import akka.event.Logging
import akka.event.LoggingAdapter

class PublisherActor : AbstractActor() {


    private val log = Logging.getLogger(context.system, this)

    class TriggerEmit//

    override fun createReceive(): AbstractActor.Receive {
        return receiveBuilder()
                .match(TriggerEmit::class.java) { event -> context.system.eventStream().publish(Message("This is a message")) }
                .build()
    }

    companion object {
        fun props(): Props {
            return Props.create(PublisherActor::class.java) { PublisherActor() }
        }
    }
}//
